### PR TITLE
boards: rv32m1_vega: Enable arduino spi instance and spi loopback test

### PR DESCRIPTION
**Pull request by MaureenHelm**
Enables the arduino spi instance, configures associated pinmuxes, and enables the spi loopback test on the rv32m1_vega board

### CODE REVIEW

**Review by dleach02 — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by galak — APPROVED** [review]

File: tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2018, NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_0"
+CONFIG_SPI_LOOPBACK_SLAVE_NUMBER=2
+CONFIG_DEBUG_OPTIMIZATIONS=y
```
Comment (galak) [inline tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf]: is CONFIG_DEBUG_OPTIMIZATIONS needed?

Comment (MaureenHelm) [reply tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf]: I tried removing it, but it caused a reset somewhere in the test.

Comment (andrewboie) [reply tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf]: I think a bug needs to filed on this

Comment (karstenkoenig) [reply tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf]: The commit message says 
> ... and turned on debug configurations to debug effectively and from excess code being optimized out.
Maybe Jose Manuel has an idea an can chime in? Sounds like something looks unreachable for the compiler and gets optimized out, kind of disconcerting

**Review by karstenkoenig — APPROVED** [review]
Loopback test works on my vegaboard, nice :-) The required debug flags are kind of strange though

**Review by dbkinder — APPROVED** [review]

**Comment by karstenkoenig** [comment]
I added a bug with me as assignee to try and root out the issue
https://github.com/zephyrproject-rtos/zephyr/issues/20589

**Comment by carlescufi** [comment]
Please rebase

**Comment by MaureenHelm** [comment]
> Please rebase

Done

**Pull request merged**